pyjetty/alice_analysis/analysis/user/substructure/run_analysis.py

```diff
@@ -637,6 +669,27 @@ def plot_systematic_uncertainties(self, jetR, obs_label, obs_setting, grooming_s
     h_total.Write()
     fSystematics.Close()
 
+  #----------------------------------------------------------------------
+  # Returns truncated 1D histogram from bins [1, ..., maxbin] inclusive
+  # (Note this uses 1-indexed bin number to comply with ROOT)
+  #----------------------------------------------------------------------
+  def truncate_hist(self, h, maxbin, new_name):
+    length = h.GetNbinsX()
+    if maxbin == None or maxbin == length:
+      h.SetNameTitle(new_name, new_name)
+      return h
+    elif maxbin > length:
+      raise ValueError("Max bin number {} larger than histogram size {}".format(maxbin, length))
+    elif maxbin < 1:
+      raise ValueError("Max bin number {} cannot be less than 1".format(maxbin))
+
+    bin_edges = [h.GetXaxis().GetBinLowEdge(i) for i in range(1, maxbin+2)]
+    h_new = ROOT.TH1D(new_name, new_name, maxbin, array('d', bin_edges))
+    for i in range(1, maxbin+1):
+      h_new.SetBinContent(i, h.GetBinContent(i))
+      h_new.SetBinError(i, h.GetBinError(i))
+    return h_new
```

Comment:
I think before returning you should do:

```
if h.GetSumw2(): 
    h_new.Sumw2()
```

But if you could think carefully to make sure that would be good :)

Reply:
And one other question: Should we include underflow and overflow bins as well? 

Actually, why not just use https://root.cern.ch/doc/master/classTH1.html#aff6520fdae026334bf34fa1800946790?

Reply:
> Actually, why not just use https://root.cern.ch/doc/master/classTH1.html#aff6520fdae026334bf34fa1800946790?

Fantastic -- I didn't realize Rebin could handle this. Fixed!

Reply:
Great, thanks! Yea ROOT can do variable rebinning for 1D, but not multiple dimensions (hence a bunch of manual rebinning code in our utils)